### PR TITLE
Implement connect / disconnect for clients

### PR DESCRIPTION
**Pull request by genuaboro**

### CODE REVIEW

File: t/client-connect.t [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    # Use the real signal handler to interrupt the OPC UA server.
+    # This is not signal safe, best effort is good enough for a test.
+    my $sigact = POSIX::SigAction->new(\&handler);
+    sigaction(SIGALRM, $sigact);
+    alarm(1);
+
+    # run server and stop after one second
+    $s->run($running);
+
+    exit 0;
```
Comment (bluhm) [inline t/client-connect.t]: When exiting from a forked child, use POSIX::_exit.  exit may execute cleanup code twice, once in the parent and one in the child.  This has very strange effects with END blocks.

File: t/client-connect.t [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    my $running = 1;
+    sub handler {
+	$running = 0;
+    }
+
+    # Perl signal handler only works between perl statements.
+    # Use the real signal handler to interrupt the OPC UA server.
+    # This is not signal safe, best effort is good enough for a test.
+    my $sigact = POSIX::SigAction->new(\&handler);
+    sigaction(SIGALRM, $sigact);
```
Comment (bluhm) [inline t/client-connect.t]: Pleade check return codes.  Something like "or die()" would be fine.

File: t/client-connect.t [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+is($r, STATUSCODE_GOOD, "client connected");
+
+is($c->getState, CLIENTSTATE_SESSION, "client state connected");
+
+$r = $c->disconnect();
+is($r, STATUSCODE_GOOD, "client disconnected");
+
+is($c->getState, CLIENTSTATE_DISCONNECTED, "client state disconnected");
+
+waitpid $pid, 0;
```
Comment (bluhm) [inline t/client-connect.t]: Check $? to see that child did run properly.

File: lib/OPCUA/Open62541.pm [hunk 4]
```diff
@@ -270,8 +270,19 @@ my @statuscodes = qw(
...
+    CLIENTSTATE_SECURECHANNEL
+    CLIENTSTATE_SESSION
+    CLIENTSTATE_SESSION_DISCONNECTED
+    CLIENTSTATE_SESSION_RENEWED
+);
+
 our %EXPORT_TAGS = (
-    'all' => [ @limits, @statuscodes ],
+    'all' => [ @limits, @statuscodes, @clientstates ],
+    'clientstates' => [ @clientstates ],
```
Comment (bluhm) [inline lib/OPCUA/Open62541.pm]: I called the key in singular, see 'statuscode'.  Use 'clientstate' for consistency.  Plural s for arrays is fine.

File: t/client-connect.t [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+my $c = OPCUA::Open62541::Client->new();
+ok($c, "client");
+
+my $cc = $c->getConfig();
+ok($c, "config client");
+
+$r = $cc->setDefault();
+is($r, STATUSCODE_GOOD, "default client config");
+
+
```
Comment (bluhm) [inline t/client-connect.t]: Two empty lines, one is enough.

File: t/client-connect.t (hunk 3 above)
Comment (genuaboro) [reply t/client-connect.t]: The check for $? is at the end.

**Pull request merged**